packages/plugin-hardhat/src/defender/utils.ts

```diff
@@ -35,6 +36,20 @@ export async function getNetwork(hre: HardhatRuntimeEnvironment): Promise<Networ
   const chainId = hre.network.config.chainId ?? (await getChainId(provider));
   const network = fromChainId(chainId);
   if (network === undefined) {
```

Comment:
Same as [here](https://github.com/OpenZeppelin/defender-deploy-client-cli/pull/7#discussion_r1516524208)

Reply:
Updated to support a `network` field in `defender` section of Hardhat config. If set, the network name must match, and this is used to disambiguate if there are multiple networks with the same chainId.